how-abstract-works/abstract-global-wallet/architecture.mdx

```diff
@@ -0,0 +1,118 @@
+    From this mnemonic phrase, a public-private key pair is derived.
+  </Step>
+  <Step title="Private Key Sharding">
+    The private key is sharded (split) into 3 parts and stored in 3 different
+    locations to ensure security and recovery mechanisms.
+  </Step>
+</Steps>
+
+#### Private Key Sharding
+
+The generated private key is split into 3 shards using
+[Shamir's Secret Sharing](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing)
+algorithm and stored in 3 different locations.
+**2 out of 3** shards are required to reconstruct the private key.
+
+The three shards are:
+
+1. **Device Share**: This shard is stored on the user's device. In a browser environment,
+   it is stored inside the local storage of the Privy iframe.
+
+2. **Auth Share**: This shard which is encrypted and stored by Privy&rsquo;s servers. It is retrieved
+   when the user logs in with their original login method.
+
+3. **Recovery Share**: This shard is stored in a back up location of the user&rsquo;s choice, typically
+   a cloud storage account such as Google Drive or iCloud.
+
+#### How Shards are Combined
+
+To reconstruct the private key, the user must have access to **2 out of 3** shards. This can be a combination
+of any two shards, the most common being the **Device Share** and **Auth Share**.
+
+- **Device Share** + **Auth Share**: This is the typical flow; the user authenticates with the Privy server
+  using their original login method (e.g. social account) on their device and the auth share is decrypted.
+
+- **Device Share** + **Recovery Share**: If the Privy server is offline or the user has lost access to their
+  original login method (e.g. they no longer have access to their social account), they can use the recovery
+  share to reconstruct the private key.
+
+- **Auth Share** + **Recovery Share**: If the user wants to access their account from a new device,
+  a new device share can be generated by combining the auth share and recovery share.
+
+### Smart Contract Wallet Deployment
+
+Once an EOA wallet is generated, the public key is provided to a
+[smart contract wallet](/how-abstract-works/native-account-abstraction/smart-contract-wallets) deployment.
+The smart contract wallet is deployed and the EOA wallet is added as an authorized signer to the wallet during
+the initialization process.
+
+As all accounts on Abstract are smart contract accounts,
+(see [native account abstraction](/how-abstract-works/native-account-abstraction)),
+the smart contract wallet is treated as a first-class citizen when interacting with the Abstract ecosystem.
+Users can use their account going forward to interact with any application (including yours) built on Abstract.
+
+The smart contract wallet that is deployed is a modified fork of
+[Clave](https://github.com/getclave/clave-contracts) customized to have an `secp256k1`
+signer by default to support the Privy Embedded Wallet _(as opposed to the default `secp256r1` signer in Clave)_.
```

Comment:
We can also mention that it has custom validation logic to verify EIP712 signatures.

I ended up writing my own validator since I didn't like the default EIP-191 behavior.